src/Internal/UriValidation/UriValidation.h

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+
+namespace AppInstaller::UriValidation
+{
+    // The decision made based on the Uri validation.
+    enum class UriValidationDecision
+    {
+        Allow,
+        Block,
+    };
+
+    // The result of a Uri validation.
+    class UriValidationResult
+    {
+    private:

```

Comment:
Nit: In most other class definitions for this WinGet, the privates and member variables are defined at the bottom of the class; I'm not sure it this is a style convention or just happenstance